x-pack/plugins/lens/public/app_plugin/shared/edit_on_the_fly/types.ts

```diff
@@ -37,22 +36,22 @@ export interface EditConfigPanelProps {
   visualizationMap: VisualizationMap;
   datasourceMap: DatasourceMap;
   /** The attributes of the Lens embeddable */
-  attributes: TypedLensByValueInput['attributes'];
+  attributes: TypedLensSerializedState['attributes'];
   /** Callback for updating the visualization and datasources state.*/
   updatePanelState: (
     datasourceState: unknown,
     visualizationState: unknown,
-    visualizationType?: string
+    visualizationId?: string
```

Comment:
That argument is used by the Unified Histogram by forwarding a `visualizationId` provided by the suggestion API.
Effectively in Lens the `visualizationType` and `visualizationId` have been used in multiple places interchangeably (it is possible to find multiple occurrencies of assignment of `visualizationId`/`activeId` to the `visualizationType`).

As far as my understanding the type and id diverge from each other only in case of mixed XY chart types.

In that specific line I've just aligned the argument name with its intended scope.

Reply:
I've created an issue to improve this https://github.com/elastic/kibana/issues/199604